### ISSUE TITLE
Testings with the Repository

### ISSUE THREAD

**Issue by heitorrapela** [vlkniaz/ThermalGAN#12]
To download the training set, you need to ask permission on the http://zefirus.org/ThermalGAN page, but it is not working right now, so I sent a message to the zefirus team. The links that they provided when you asked for the dataset are still working, but as I had to ask for permission, I will wait for them to fix the ThermalGAN page,

To download ThermalWorld test set:

`./datasets/download_thermalworld_dataset.sh ThermalWorld`

I will keep this issue open while I am testing the repository.

**Comment by ShirGrinblat** [comment]
@heitorrapela Any news? I'm trying to download the dataset without any success!

**Comment by heitorrapela** [comment]
Sorry, but I do not have news, but you can download with the following links (command for Linux: wget -r link):

- http://zefirus.org/datasets/ThermalWorld_VOC_v1_0.zip
- https://www.dropbox.com/s/d35d3q9n055a1pc/ThermalWorld_VOC_v1_0.zip?dl=0
- http://zefirus.org/datasets/ThermalWorld_ReID_train_v3_0.zip
- https://www.dropbox.com/s/mvc0euwqmjp181l/ThermalWorld_ReID_train_v3_0.zip?dl=0

**Comment by phflot** [comment]
> Sorry, but I do not have news, but you can download with the following links (command for Linux: wget -r link):
> 
> * http://zefirus.org/datasets/ThermalWorld_VOC_v1_0.zip
> * https://www.dropbox.com/s/d35d3q9n055a1pc/ThermalWorld_VOC_v1_0.zip?dl=0
> * http://zefirus.org/datasets/ThermalWorld_ReID_train_v3_0.zip
> * https://www.dropbox.com/s/mvc0euwqmjp181l/ThermalWorld_ReID_train_v3_0.zip?dl=0

I am trying to download the dataset but all the links don't seem to work anymore. Do you have an updated link for the dataset that you could share?

**Comment by heitorrapela** [comment]
I had it saved here on my computer. These are the original files from the authors I uploaded to in drive.

[ThermalWorld_VOC_v1_0.zip](https://drive.google.com/file/d/1ZSfxTyXbgw22Sh-HGJ26ba_-XPNs_IJV/view?usp=share_link)

[ThermalWorld_ReID_train_v3_0.zip](https://drive.google.com/file/d/1XIc_i3mp4xFlDJ_S5WJYMJAHq107irPI/view?usp=share_link)

**Comment by phflot** [comment]
Thank you very much!

**Comment by jcgit786** [comment]
@heitorrapela Do you have the pre-trained model for the ThermalWorld Dataset too by any chance?

**Comment by jcgit786** [comment]
@phflot Do you have the pre-trained model for the ThermalWorld Dataset too by any chance?

**Comment by phflot** [comment]
> @phflot Do you have the pre-trained model for the ThermalWorld Dataset too by any chance?

I ended up not using the ThermalWorld dataset, because my application requires a higher level of cross-modal alignment. If you are working on faces, you could have a look at the Sejong face dataset: https://github.com/ahmadmobeen/SyntheticDisguiseFaceDatabase

**Comment by heitorrapela** [comment]
@jcgit786 sorry but I dont have.

**Comment by NQHuy1905** [comment]
@heitorrapela have you tried inference model with your data, i tried but it return error because input of model is 4 channel while RGB only have 3 channel. Do you know how to fix that?

**Comment by franckmarineai** [comment]
> @heitorrapela have you tried inference model with your data, i tried but it return error because input of model is 4 channel while RGB only have 3 channel. Do you know how to fix that?

in test.py change 'for i, data in enumerate(dataset)' -> 'for i, data in enumerate(dataloader)'
if you get an error on getting 'img_path'  uncomment the following line in 'models/infagan.py'
self.image_paths == [....]

**Comment by Tayeba** [comment]
> I had it saved here on my computer. These are the original files from the authors I uploaded to in drive.
> 
> [ThermalWorld_VOC_v1_0.zip](https://drive.google.com/file/d/1ZSfxTyXbgw22Sh-HGJ26ba_-XPNs_IJV/view?usp=share_link)
> 
> [ThermalWorld_ReID_train_v3_0.zip](https://drive.google.com/file/d/1XIc_i3mp4xFlDJ_S5WJYMJAHq107irPI/view?usp=share_link)
 @heitorrapela  Thanks for uploading this data on the drive. I was able to download, but is this the complete dataset?
Were you able to train and test the model using the above dataset?

**Comment by heitorrapela** [comment]
Hello, if you check the ThermalWorld_VOC_v1_0.zip, there is the ThermalGAN.py with the code that the authors used to decode the thermal '.npy' imgs to '.png' imgs,

I made a [script](https://github.com/vlkniaz/ThermalGAN/files/10912100/script.zip) to do it that maybe is going to help everyone here, the base of the code is the same one provided by the authors to decode the image, but I do know if it helps anything. Then, with the thermal images, you can use them for anything that you want. In my case, I trained some cycle gans to translate between RGB and thermal, but it was a long time ago, and I can not help with that. I could not upload directly the .py here, so I compressed it.
